src/Main.elm

```diff
@@ -1,47 +1,176 @@
 module Main exposing (..)
 
 import Browser
-import Html exposing (Html, text, div, h1, img)
-import Html.Attributes exposing (src)
+import Html exposing (..)
+import Html.Attributes exposing (class, href)
+import Html.Events exposing (..)
+import Http
+import Json.Decode as Decode exposing (Decoder, andThen, field, int, list, string, succeed)
+import Json.Decode.Pipeline exposing (required)
+import RemoteData exposing (RemoteData, WebData)
+import Debug exposing (log)
 
 
 ---- MODEL ----
 
 
+type alias StoryId =
+  Int
+
+type alias Story =
+    { title: String
+    , url: String
+    }
 type alias Model =
-    {}
+    { stories : WebData (List Story)
+    , storyIds : WebData ( List StoryId)
+    }
+
+
+emptyModel : Model
+emptyModel =
+    { stories = RemoteData.Loading
+    ,storyIds = RemoteData.Loading
+    }
 
 
 init : ( Model, Cmd Msg )
 init =
-    ( {}, Cmd.none )
+    ( emptyModel, Cmd.none )
 
 
 
 ---- UPDATE ----
 
 
 type Msg
-    = NoOp
+    = HandleFetchedStoryIds (WebData (List StoryId))
+    | HandleFetchedStory ( WebData (List Story))
 
+newsDecoder : Decoder Story
+newsDecoder =
+    Decode.succeed Story
+        |> required "title" string
+        |> required "url" string
 
-update : Msg -> Model -> ( Model, Cmd Msg )
-update msg model =
-    ( model, Cmd.none )
 
 
 
+
+getStory : Int -> Cmd Msg
+getStory id =
+    Http.get
+        { url = "https://hacker-news.firebaseio.com/v0/item/" ++ ( String.fromInt id) ++ ".json"
+
+        , expect =
+            list newsDecoder
+                |> Http.expectJson ( RemoteData.fromResult >> HandleFetchedStory )
+        }
+
+getStoryId : Cmd Msg
+getStoryId =
+    Http.get
+        { url = "https://hacker-news.firebaseio.com/v0/topstories.json"
+        , expect =
+            list int
+                |> Http.expectJson (RemoteData.fromResult >>  HandleFetchedStoryIds  )
+
+        }
+
+
+
+update : Msg -> Model -> ( Model, Cmd Msg )
+update msg model =
+    case msg of
+        HandleFetchedStoryIds webdata ->
+            let
+                idList =
```

Comment:
@anvmn 
Here I have the decoded list of ids and here I am able to log the list. So, now I have to create a map function and I will map through this list. How do I access this list?